net-p2p/eiskaltdcpp/eiskaltdcpp-2.2.10.ebuild

```diff
@@ -0,0 +1,149 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="5"
+
+PLOCALES="be bg cs de el en es eu fr hu it pl pt_BR ru sk sr@latin sr sv_SE uk vi zh_CN"
+
+inherit cmake-utils eutils l10n fdo-mime gnome2-utils
+[[ ${PV} = *9999* ]] && inherit git-r3
+
+DESCRIPTION="Qt based client for DirectConnect and ADC protocols, based on DC++ library"
+HOMEPAGE="https://github.com/eiskaltdcpp/eiskaltdcpp"
+
+LICENSE="GPL-2 GPL-3"
+SLOT="0"
+IUSE="cli daemon dbus +dht +emoticons examples -gtk idn -javascript json libcanberra libnotify lua +minimal pcre +qt4 sound spell sqlite upnp -xmlrpc"
+
+REQUIRED_USE="
+	cli? ( ^^ ( json xmlrpc ) )
+	emoticons? ( || ( gtk qt4 ) )
+	dbus? ( qt4 )
+	javascript? ( qt4 )
+	json? ( !xmlrpc )
+	libcanberra? ( gtk )
+	libnotify? ( gtk )
+	spell? ( qt4 )
+	sound? ( || ( gtk qt4 ) )
+	sqlite? ( qt4 )
+"
+
+if [[ ${PV} != *9999* ]]; then
+	SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+else
+	EGIT_REPO_URI="https://github.com/${PN}/${PN}.git"
+	KEYWORDS=""
+fi
+
+RDEPEND="
+	app-arch/bzip2
+	>=dev-libs/boost-1.38:=
+	<dev-libs/openssl-1.1:0=
+	sys-apps/attr
+	sys-libs/zlib
+	virtual/libiconv
+	virtual/libintl
+	idn? ( net-dns/libidn )
+	lua? ( >=dev-lang/lua-5.1:= )
+	pcre? ( >=dev-libs/libpcre-4.2 )
+	upnp? ( >=net-libs/miniupnpc-1.6 )
+	cli? (
+		>=dev-lang/perl-5.10
+		virtual/perl-Getopt-Long
+		dev-perl/Data-Dump
+		dev-perl/Term-ShellUI
+		json? ( dev-perl/JSON-RPC )
+		xmlrpc? ( dev-perl/RPC-XML )
+	)
+	daemon? ( xmlrpc? ( >=dev-libs/xmlrpc-c-1.19.0[abyss,cxx] ) )
+	gtk? (
+		x11-libs/pango
+		x11-libs/gtk+:3
+		>=dev-libs/glib-2.24:2
+		x11-themes/hicolor-icon-theme
+		libcanberra? ( media-libs/libcanberra )
+		libnotify? ( >=x11-libs/libnotify-0.4.1 )
+	)
+	qt4? (
+		>=dev-qt/qtcore-4.6.0:4
+		>=dev-qt/qtgui-4.6.0:4
+		dbus? ( >=dev-qt/qtdbus-4.6.0:4 )
+		javascript? (
+			dev-qt/qtscript:4
+			x11-libs/qtscriptgenerator
+		)
+		spell? ( app-text/aspell )
+		sqlite? ( dev-qt/qtsql:4[sqlite] )
+	)
+"
+DEPEND="${RDEPEND}
+	sys-devel/gettext
+	virtual/pkgconfig
+"
+DOCS=( AUTHORS ChangeLog.txt )
+
+pkg_pretend() {
+	if [[ ${MERGE_TYPE} != binary ]]; then
+		[[ $(gcc-major-version) -lt 4 ]] || \
+				( [[ $(gcc-major-version) -eq 4 && $(gcc-minor-version) -le 6 ]] ) \
```

Comment:
The subshell (`( ... )`) is absolutely unnecessary here. It would probably make sense to put this in a single `[[ ... ]]` or use `{ ...; }`… however, I think it makes even more sense to just drop this check because we don't really support gcc that old anyway and worst things than this will explode ;-).